### ISSUE TITLE
Feature Request: Break jasmine/valid-expect into separate rules

### ISSUE THREAD

**Issue by lukas-gitl** [tlvince/eslint-plugin-jasmine#144]
We would like to suppress only some warnings for this rule. 

E.g. we use
(1) dynamically evaluated expect and
(2) custom messaging with `expect(..., "custom message").to.equal(...)`
However the other warnings are still valid for us.

By breaking the rule into separate rules we would obtain the desired fine control over the warnings.

**Comment by DianaSuvorova** [comment]
@lukas-gitl could you provide an example for `dynamically evaluated expect` case.
And for custom message for expect. Is it part of core jasmine library or some extension? Can you provide a link to docs?

**Comment by lukas-gitl** [comment]
@DianaSuvorova 
(1) sure
```javascript
const handleDynamicExpect = (target, check) => {
  let result = 0;
  Object.keys(check).forEach((key) => {
    let targetInternal = target;
    const value = check[key];
    const keys = key.split(".");
    const lastKey = keys.pop();
    keys.forEach((p) => {
      targetInternal = targetInternal[p];
    });
    if (typeof value !== 'object' || value instanceof Array) {
      // check if regexp or not and execute the check
      const isRegex = typeof value === 'string' && value.indexOf("^") === 0;
      targetInternal[lastKey](isRegex ? new RegExp(value, "i") : value);
      result += 1;
    } else {
      result += handleDynamicExpect(targetInternal[lastKey], value);
    }
  });
  return result;
};
```
called with e.g.
```javascript
handleDynamicExpect(
  expect("test"),
  [
    {
      "to.deep.equal": [
        "Rule on Cool Down."
      ]
    }
  ]
);
```
(2) not officially documented, but it seems to work great

We just want a way to suppress some warnings for this rule while keeping the others. Alternative is to disable it and pull it in as a custom rule. Which might actually be what we end up with...

**Comment by DianaSuvorova** [comment]
I would agree, technically `valid-expect` rule does 2 checks in one: `expect-has-one-argument`, `no-expect-without-matcher`.  I think breaking this into 2 rules is a good idea. We would need to deprecate `valid-expect` and introduce 2 new rules. @burabure , thoughts?

**Comment by lukas-gitl** [comment]
This is what we are using now (custom rule):

```javascript
module.exports = context => ({
  CallExpression: (node) => {
    if (node.callee.name === 'expect') {
      // checking "expect()" arguments
      if (node.arguments.length > 1) {
        if (node.arguments.length === 2) {
          if (['Literal', 'TemplateLiteral'].indexOf(node.arguments[1].type) === -1) {
            context.report({
              message: 'Expected second argument to be debug string',
              node
            });
          }
        } else {
          context.report({
            message: 'More than one argument passed to expect()',
            node
          });
        }
      } else if (node.arguments.length === 0) {
        context.report({
          message: 'No arguments passed to expect()',
          node
        });
      }
    }
  },

  // nothing called on "expect()"
  'CallExpression:exit': (node) => {
    if (node.callee.name === 'expect' && node.parent.type === 'ExpressionStatement') {
      context.report({
        message: 'Nothing called on expect()',
        node
      });
    }
  }
});
```

**Comment by DianaSuvorova** [comment]
This makes sense to me. Let me look into proper way of deprecating rules and replacing it with new one(s).

**Comment by lukas-gitl** [comment]
Sweet, thank you!

**Comment by lukas-gitl** [comment]
Just curious, is https://github.com/DianaSuvorova/eslint-plugin-jasmine/blob/e647aff743c59b297766927fc5cb3b6796c10800/lib/rules/expect-matcher.js supposed to now allow for our use case: https://github.com/tlvince/eslint-plugin-jasmine/issues/144#issuecomment-334016490? Because it doesn't seem to work.
